coral-common/src/main/java/com/linkedin/coral/common/transformers/OperatorRenameSqlCallTransformer.java

```diff
@@ -0,0 +1,29 @@
+/**
+ * Copyright 2023 LinkedIn Corporation. All rights reserved.
+ * Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.coral.common.transformers;
+
+import org.apache.calcite.sql.SqlCall;
+import org.apache.calcite.sql.SqlNodeList;
+import org.apache.calcite.sql.SqlOperator;
+import org.apache.calcite.sql.parser.SqlParserPos;
+
+
+public class OperatorRenameSqlCallTransformer extends SourceOperatorMatchSqlCallTransformer {
+  private SqlOperator sourceOperator;
+  private String targetOpName;
+
+  public OperatorRenameSqlCallTransformer(SqlOperator sourceOperator, int numOperands, String targetOpName) {
```

Comment:
I think `sourceOperator` can be replaced by the source operator name directly, we can use `sqlCall.getOperator().getReturnTypeInference()` to get the return type for the following transformation part. Modified it to be [OperatorRenameSqlCallTransformer(String sourceOpName, int numOperands, String targetOpName)](https://github.com/linkedin/coral/pull/351/files#diff-ef4e434c5fc6a421de8fdb0f9b3b73e5dc4474000fc0d87aecf979d372a44fb6R19). cc: @wmoustafa

Reply:
As `CoralRegistryOperatorRenameSqlCallTransformer` inherits `OperatorRenameSqlCallTransformer`  and it depends on the SqlOperator obtained from `StaticHiveFunctionRegistry` as follows. It is better not to remove `souceOperator` out of `OperatorRenameSqlCallTransformer`
```java
public class CoralRegistryOperatorRenameSqlCallTransformer extends OperatorRenameSqlCallTransformer {
  private static final StaticHiveFunctionRegistry HIVE_FUNCTION_REGISTRY = new StaticHiveFunctionRegistry();

  public CoralRegistryOperatorRenameSqlCallTransformer(String sourceOpName, int numOperands, String targetOpName) {
    super(HIVE_FUNCTION_REGISTRY.lookup(sourceOpName).iterator().next().getSqlOperator(), numOperands, targetOpName);
  }
}
```

Reply:
I think we can modify it to be:
```
    super(HIVE_FUNCTION_REGISTRY.lookup(sourceOpName).iterator().next().getSqlOperator().getName(), numOperands, targetOpName);

```

Reply:
Talked with @ljfgem offline, we agree to address this comment later.